pkg/containers/containers.go

```diff
@@ -246,6 +206,52 @@ func (c *Containers) CgroupRemove(cgroupId uint64) {
 	c.deleted = append(c.deleted, cgroupId)
 }
 
+func (c *Containers) CgroupMkdir(cgroupId uint64, subPath string) (CgroupInfo, error) {
+	curTime := time.Now()
+	// first try to find this new cgroup in cgroup fs
+	path, err := getCgroupPath(c.cgroupMP, cgroupId, subPath)
+	if err == nil {
+		// cgroup entry found in cgroupfs
+		var stat syscall.Stat_t
+		if err := syscall.Stat(path, &stat); err == nil {
+			return c.CgroupUpdate(cgroupId, path, time.Unix(stat.Ctim.Sec, stat.Ctim.Nsec))
+		}
+	}
+
+	// if no entry found, container may have already exited
+	// update map with existing data (ctime becomes an estimation only)
+	return c.CgroupUpdate(cgroupId, subPath, curTime)
+}
+
+func getCgroupPath(rootDir string, cgroupId uint64, subPath string) (string, error) {
+	entries, err := os.ReadDir(rootDir)
+	if err != nil {
+		return "", err
+	}
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			continue
+		}
+		entryPath := filepath.Join(rootDir, entry.Name())
+		if strings.HasSuffix(entryPath, subPath) {
```

Comment:
In my system I'm getting empty subPath (lots of), making this to always be executed in such cases. So the logic works, but you should check why subPaths are empty to reduce execution overhead (and perhaps not run this logic if subPath is empty). 

Check it out:

```
subPath:
entryPath: /sys/fs/cgroup/system.slice/acpid.socket
subPath:
entryPath: /sys/fs/cgroup/system.slice/alsa-restore.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/avahi-daemon.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/avahi-daemon.socket
subPath:
entryPath: /sys/fs/cgroup/system.slice/binfmt-support.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/blk-availability.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/boot.mount
subPath:
entryPath: /sys/fs/cgroup/system.slice/console-setup.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/containerd.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/cpufrequtils.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/cron.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/dbus.service
subPath:
entryPath: /sys/fs/cgroup/system.slice/dbus.socket
subPath:
entryPath: /sys/fs/cgroup/system.slice/dm-event.socket
subPath:
entryPath: /sys/fs/cgroup/system.slice/docker-cd93f6d44215e34daa9688e7998beced9dae9424994bd2ca792c76c22288be6f.scope
subPath:
entryPath: /sys/fs/cgroup/system.slice/docker-cd93f6d44215e34daa9688e7998beced9dae9424994bd2ca792c76c22288be6f.scope
subPath: /system.slice/docker-cd93f6d44215e34daa9688e7998beced9dae9424994bd2ca792c76c22288be6f.scope
```

Reply:
This function is also invoked from GetCgroupInfo as follows:
`getCgroupPath(c.cgroupMP, cgroupId, "")`
Which explains why you get the empty subPath - this is expected.

> `if subPath != "" && strings.HasSuffix(entryPath, subPath) {`

Will do

Reply:
@rafaeldtinoco how often you see an empty subPath in your environment?
This shuold only happen when cgroup does not exist in the map, which shouldn't happen too often (Actually, almost never)

Reply:
Ok, there was indeed an issue here.
cgroup_mkdir event wasn't processed in the case where container filter is set (I guess you did your checks with container filters set, right?)
The problem was that we filter for "false container poitives" and ignore the event in such cases. But cgroup_mkdir event is not a container event (but happens before the container starts), so we didn't get it.

I'll fix it in this PR, good that you found it